core/arch/x86/decode_fast.c

```diff
@@ -1385,19 +1385,9 @@ decode_cti(dcontext_t *dcontext, byte *pc, instr_t *instr)
      * Rare enough that we do a full decode on an opcode match.
      */
     if ((byte0 == 0xdd && ((byte1 >> 3) & 0x7) == 6) /* dd /6 == OP_fnsave */ ||
-        (byte0 == 0xd9 && ((byte1 >> 3) & 0x7) == 6) /* d9 /6 == OP_fnstenv */) {
-        if (decode(dcontext, start_pc, instr) == NULL)
-            return NULL;
-        else
-            return (start_pc + sz);
-    } else if (byte0 == 0x0f && byte1 == 0xae) {
-        int opc_ext;
-        byte byte3 = *(pc + 2);
-        opc_ext = (byte3 >> 3) & 0x7;
-        if (opc_ext == 0 ||  /* 0f ae /0 == OP_fxsave */
-            opc_ext == 4 ||  /* 0f ae /4 == OP_xsave */
-            opc_ext == 6) {  /* 0f ae /6 == OP_xsaveopt */
-        }
+        (byte0 == 0xd9 && ((byte1 >> 3) & 0x7) == 6) /* d9 /6 == OP_fnstenv */ ||
+        (byte0 == 0x0f && byte1 == 0xae) /* includes fxsave, xsave, xsaveopt */ ||
```

Comment:
IMHO we should include further bit checks for both 0fae and 0fc7 to avoid the cost of decode() which is substantial: it is very expensive compared to a few bit checks.  (Looks like the 0fae ones were poorly done before and not even hooked up...)  We do not want to pay for decode() on instructions we don't care about on this path.